modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/synonyms/SynonymsManagementAPIService.java

```diff
@@ -26,6 +34,32 @@ public class SynonymsManagementAPIService {
     public static final String SYNONYMS_INDEX = ".synonyms";
     public static final String SYNONYMS_ORIGIN = "synonyms";
 
+    private final Client client;
+
+    public SynonymsManagementAPIService(Client client) {
+        // TODO Should we set an OriginSettingClient? We would need to check the origin at AuthorizationUtils if we set an
+        this.client = client;
```

Comment:
Should we set an `OriginSettingClient` for these actions? Or are they used just from XPack modules? We would need to check the origin at `AuthorizationUtils` if we do

Reply:
I think we need client with origin. We can work on authorization in subsequent PRs, for now we can put just a TODO item to address